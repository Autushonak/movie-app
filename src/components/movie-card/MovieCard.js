import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { colors } from '../../constants/colors';

const MovieCardStyled = styled.div`
  display: flex;
  flex-direction: column;
`;
const MoviePoster = styled.img`
  width: 100%;
  height: 550px;
  margin-bottom: 10px;
`;

const MovieInfo = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-direction: row;
`;

const MovieName = styled.h3`
  color: ${colors.WHITE};
  margin: 10px 0;
`;

const MovieYearBox = styled.div`
  border: 1px solid ${colors.WHITE};
  padding: 3px 10px;
  border-radius: 4px;
`;

const MovieYear = styled.h3`
  color: ${colors.WHITE};
  margin: 0;
`;

const GenreBlock = styled.div`
  color: ${colors.WHITE};
`;

const Genre = styled.h5`
  color: ${colors.WHITE};
  margin: 0;
`;

const MovieCard = ({ url, title, year, genre }) => (
  <MovieCardStyled>
    <MoviePoster src={url}></MoviePoster>
    <MovieInfo>
      <MovieName>{title}</MovieName>
      <MovieYearBox>
        <MovieYear>{year}</MovieYear>
      </MovieYearBox>
    </MovieInfo>
    <GenreBlock>
      <Genre>{genre}</Genre>
    </GenreBlock>
  </MovieCardStyled>
);

MovieCard.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string,
  year: PropTypes.string,
  genre: PropTypes.string,
};

export default MovieCard;
