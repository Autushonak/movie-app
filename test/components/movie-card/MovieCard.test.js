import 'jest-styled-components';
import React from 'react';
import { mount } from 'enzyme';
import MovieCard from '../../../src/components/movie-card/MovieCard';

describe('MovieCard ', () => {
  test('should be rendered correctly', () => {
    const component = mount(<MovieCard />);
    expect(component).toMatchSnapshot();
  });
});
