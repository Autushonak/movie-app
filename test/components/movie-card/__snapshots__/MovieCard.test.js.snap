// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard  should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  height: 550px;
  margin-bottom: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  color: #FFFFFF;
  margin: 10px 0;
}

.c4 {
  border: 1px solid #FFFFFF;
  padding: 3px 10px;
  border-radius: 4px;
}

.c5 {
  color: #FFFFFF;
  margin: 0;
}

.c6 {
  color: #FFFFFF;
}

.c7 {
  color: #FFFFFF;
  margin: 0;
}

<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MovieCard>
      <Link>
        <LinkAnchor
          href=""
          navigate={[Function]}
        >
          <a
            href=""
            onClick={[Function]}
          >
            <styled.div>
              <div
                className="c0"
              >
                <styled.img>
                  <img
                    className="c1"
                  />
                </styled.img>
                <styled.div>
                  <div
                    className="c2"
                  >
                    <styled.h3>
                      <h3
                        className="c3"
                      />
                    </styled.h3>
                    <styled.div>
                      <div
                        className="c4"
                      >
                        <styled.h3>
                          <h3
                            className="c5"
                          />
                        </styled.h3>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="c6"
                  >
                    <styled.h5>
                      <h5
                        className="c7"
                      />
                    </styled.h5>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </a>
        </LinkAnchor>
      </Link>
    </MovieCard>
  </Router>
</BrowserRouter>
`;
